spec/coordinates_spec.rb

```diff
@@ -0,0 +1,31 @@
+require_relative "../lib/coordinates"
+
+RSpec.describe Coordinates do
+  describe "#valid?" do
+    it "is true when the value is in the grid" do
+      described_class.valid_values.each do |value|
+        coordinates = described_class.new(value: value)
+
+        expect(coordinates).to be_valid
+      end
+    end
+
+    it "is false when the value is not in the grid" do
+      boundaries = %w[A0 B0 C0 A4 B4 C4 D1 D2 D3]
```

Comment:
This weekend I was reading a blog post by Justin Searls at Test Double about the right level and amount of testing. It's really good. Reading it could make these specs better. 

I am not a fan of putting a loop into a spec. It is pretty hard to get your heard around. In this case the data that generates the loop also requires a bit too much knowledge. 

I think there are real concerns in these tests that don't need to as exhaustive:
* When in a row before the first row of the grid
* When in a row after the last row in the grid
* When in a column before the first column in the grid
* When in a column after the last column of the grid

The other specs are very exhaustive, but not expressing the most important logic.

Reply:
I like that reasoning. Updating in 9c06d7236f0f90754186c91317fadaa306dc725.

I'm not sure what a good test value for before the first row is, though. What comes before A?